core/src/main/java/org/apache/metamodel/query/SelectItem.java

```diff
@@ -484,11 +498,24 @@ public StringBuilder toStringNoAlias(boolean includeSchemaInColumnPath) {
             final Object[] functionParameters = getFunctionParameters();
             if (functionParameters != null && functionParameters.length != 0) {
                 for (int i = 0; i < functionParameters.length; i++) {
-                    functionBeginning.append('\'');
-                    functionBeginning.append(functionParameters[i]);
-                    functionBeginning.append('\'');
+                    if (functionParameters[i] instanceof Column) {
+                        Column col = (Column) functionParameters[i];
+                        functionBeginning.append(getToStringColumnPrefix(includeSchemaInColumnPath, col));
+                        functionBeginning.append(col.getQuotedName());
+                    } else {
+                        if (!_function.getFunctionName().equals(FunctionType.CONCAT.getFunctionName())) {
+                            functionBeginning.append('\'');
+                        }
+                        functionBeginning.append(functionParameters[i]);
+                        if (!_function.getFunctionName().equals(FunctionType.CONCAT.getFunctionName())) {
+                            functionBeginning.append('\'');
+                        }
```

Comment:
Actually, looking at it again, how about just doing an ```if (functionParameters[i] instanceof String)``` and then only adding the quotes before and after in that case? For numbers, columns, selectitems etc. we wouldn't want any quote I think. And that does not seem to be specific to the CONCAT function at all?